File: 35-search-insert-position/35-search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let hi = nums.length-1, lo = 0;
    
    while(lo <= hi){
        let mid = Math.floor((hi+lo)/2)
        if(nums[mid] === target) return mid;
        
        if(nums[mid] > target){
            hi = mid - 1
        } else {
            lo = mid + 1
        }
    }
    return lo
};

File: 35-search-insert-position/35-search-insert-position.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search_insert(nums, target)
    lo, hi = 0, nums.length-1
    
    while lo <= hi
        mid = lo + (hi-lo) / 2
        
        return mid if nums[mid] == target
        
        if nums[mid] < target
            lo = mid+1
        else
            hi = mid-1
        end
    end
    lo
end